js/urlsearchparams.ts

```diff
@@ -0,0 +1,174 @@
+export class URLSearchParams {
+
+	private params: [string, string][] = [];
+
+	public constructor(init: string | Iterable<[string, string]> | Record<string, string> = '') {
```

Comment:
`init` may be `undefined`
``````
var u = new URLSearchParams()
undefined
u instanceof URLSearchParams
true
``````

Reply:
The `=` at the end states it's optional. It defaults to `''` to comply with [WHATWG](https://url.spec.whatwg.org/#interface-urlsearchparams).